examples/api-samples/src/browser-only/filesystem/example-filesystem-initialization.ts

```diff
@@ -17,35 +17,36 @@
 import { URI } from '@theia/core';
 import { inject, injectable, interfaces } from '@theia/core/shared/inversify';
 import { EncodingService } from '@theia/core/lib/common/encoding-service';
-import { BrowserFSInitialization, DefaultBrowserFSInitialization } from '@theia/filesystem/lib/browser-only/browserfs-filesystem-initialization';
-import { BrowserFSFileSystemProvider } from '@theia/filesystem/lib/browser-only/browserfs-filesystem-provider';
-import type { FSModule } from 'browserfs/dist/node/core/FS';
+import { OPFSInitialization, DefaultOPFSInitialization } from '@theia/filesystem/lib/browser-only/opfs-filesystem-initialization';
+import { OPFSFileSystemProvider } from '@theia/filesystem/lib/browser-only/opfs-filesystem-provider';
 
 @injectable()
-export class ExampleBrowserFSInitialization extends DefaultBrowserFSInitialization {
+export class ExampleOPFSInitialization extends DefaultOPFSInitialization {
 
     @inject(EncodingService)
     protected encodingService: EncodingService;
 
-    override async initializeFS(fs: FSModule, provider: BrowserFSFileSystemProvider): Promise<void> {
+    override async initializeFS(provider: OPFSFileSystemProvider): Promise<void> {
+        // Check whether the directory exists
         try {
-            if (!fs.existsSync('/home/workspace')) {
-                await provider.mkdir(new URI('/home/workspace'));
-                await provider.writeFile(new URI('/home/workspace/my-file.txt'), this.encodingService.encode('foo').buffer, { create: true, overwrite: false });
-                await provider.writeFile(new URI('/home/workspace/my-file2.txt'), this.encodingService.encode('bar').buffer, { create: true, overwrite: false });
-            }
-            if (!fs.existsSync('/home/workspace2')) {
-                await provider.mkdir(new URI('/home/workspace2'));
-                await provider.writeFile(new URI('/home/workspace2/my-file.json'), this.encodingService.encode('{ foo: true }').buffer, { create: true, overwrite: false });
-                await provider.writeFile(new URI('/home/workspace2/my-file2.json'), this.encodingService.encode('{ bar: false }').buffer, { create: true, overwrite: false });
-            }
+            await provider.readdir(new URI('/home/workspace'));
         } catch (e) {
-            console.error('An error occurred while initializing the demo workspaces', e);
+            console.error('An error occurred while reading the demo workspaces', e);
```

Comment:
We should check whether the error is a file not found error. Only then we should create files. Also we should not log the error as this is an expected error for any first time user.

Only other errors than the expected ones should be logged.

Reply:
Even better would be to offer some kind of "exists" method so that we don't need to go through try-catch here.

Reply:
I agree and `exists` method would definitely make sense. Unfortunately, it is not yet supported by the OPFS, but we can implement a workaround until then 👍🏼